### PR TITLE
Sample: test_train_split

### PR DESCRIPTION
**Pull request by cooper-sloan**
JIRA: MADLIB-1119

Add utility to sample test and train
data from an input table.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/154/

File: src/ports/postgres/modules/sample/test_train_split.sql_in [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+<div class="toc"><b>Contents</b>
+<ul>
+<li><a href="#strs">test train split</a></li>
+<li><a href="#examples">Examples</a></li>
+</ul>
+</div>
+
+@brief A method for independently sampling subpopulations (strata).
+
+test train split is a method for independently sampling
```
Comment (orhankislal) [inline src/ports/postgres/modules/sample/test_train_split.sql_in]: The explanation should be modified for the test-train functionality.

File: src/ports/postgres/modules/sample/test_train_split.py_in [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+def test_train_split(schema_madlib, source_table, output_table, train_proportion,
+                     test_proportion, grouping_cols, target_cols, with_replacement,
+                     separate_output_tables, **kwargs):
+    """
+    test train split function
+    Args:
+        @param source_table     Input table name.
+        @param output_table     Output table name.
+        @param proportion       The ratio of sample size to the number of
```
Comment (orhankislal) [inline src/ports/postgres/modules/sample/test_train_split.py_in]: proportion should be replaced by train_proportion and test proportion

File: src/ports/postgres/modules/sample/test_train_split.py_in [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+                                records.
+        @param grouping_cols    (Default: NULL) The columns to distinguish
+                                each strata.
+        @param target_cols      (Default: NULL) The columns to include in
+                                the output.
+        @param with_replacement (Default: FALSE) The sampling method.
+
```
Comment (orhankislal) [inline src/ports/postgres/modules/sample/test_train_split.py_in]: Missing parameter: separate_output_tables

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/158/

**Comment by iyerr3** [comment]
+1

**Comment by njayaram2** [comment]
jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/164/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/165/

**Pull request closed without merging**